rockusb/src/libusb.rs

```diff
@@ -256,6 +256,18 @@ impl Transport {
     pub fn reset_device(&mut self, opcode: ResetOpcode) -> Result<()> {
         self.handle_operation(crate::operation::reset_device(opcode))
     }
+
+    /// Get which mode the Rock device is booted.
+    pub fn boot_mode(&self) -> Result<BootMode> {
+        let version = self
+            .handle
+            .device()
+            .device_descriptor()?
+            .device_version()
+            .sub_minor()
+            & 0x1;
```

Comment:
Right this discussion reminds me why i didn't implement some bootmode type functionality; There didn't seem a reliable way to detect it..  

For reference what i do in boardswarm is sending a flash-info command and if it fails assume it's maskrom mode: 
https://github.com/sjoerdsimons/boardswarm/blob/3b7b2f9c0afd91ee3cf790166391610f15645892/boardswarm/src/rockusb.rs#L391

which seems to work well enough; though it's ofcourse in no way perfect either;

So yeah overall I'd be in favour of having a way to determine boot-mode; but if it's in the crate it should really be reliable otherwise there doesn't seem to be much point :) For your project you can ofcourse make more assumptions (e.g. if you can assume using specific SoC's/usbplug versions etc). So lets close this for now (as per your suggestion)